system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -42,7 +42,10 @@ int __timedwait_cp(volatile int *addr, int val,
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
 	int is_main_thread = emscripten_is_main_browser_thread();
-	if (is_main_thread || pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS) {
+	// cp suffix in the function name means "cancellation point", so this wait can be cancelled
+	// by the users unless current threads cancelability is set to PTHREAD_CANCEL_DISABLE
+	// which may be either done by the user of __timedwait() function.
+	if (is_main_thread || pthread_self()->canceldisable != PTHREAD_CANCEL_DISABLE) {
```

Comment:
This change is causing `'test_pthread_setcanceltype_1_1`' test to fail.

Call chain:

``pthread_mutex_lock --> __timedwait() --> __pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);``

In test there are following flags set:
```
	pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
```

It looks that this condition has to be modified.

Reply:
Updated test in 0eb64ed0e